openide.awt/src/org/openide/awt/ActionState.java

```diff
@@ -59,6 +59,9 @@
  * <li>if {@link #checkValue} is {@link #NON_NULL_VALUE}, the action is checked if and only if
  * the value is not {@code null}. 
  * <li>if the value type is an enum, its {@link Enum#name} is compared to {@link #checkValue}
+ * <li>if the value is a {@link Collection} or {@link Map}, state evaluates to the {@link Collection#isEmpty} or
+ * {@link Map#isEmpty}, respectively.
+ * <li>if the value is a {@link Number}, state will be true if value evaluates to <b>a positive integer</b>.
```

Comment:
Amazing JavaScript-like behaviour. Just slightly different conversions:
```js
> if (false) console.log("Hi"); else console.log("čehi")
čehi
> if ("") console.log("Hi"); else console.log("čehi")
čehi
> if ([]) console.log("Hi"); else console.log("čehi")
Hi
> if (3) console.log("Hi"); else console.log("čehi")
Hi
> if (-3) console.log("Hi"); else console.log("čehi")
Hi
> if (0) console.log("Hi"); else console.log("čehi")
čehi
```
but I don't think the difference matters much. Treating empty collection as `false` is certainly good idea.

Reply:
Well ... there's probably no good model with integers. And the behaviour is not "complete" enough with Collections either: some models use indexes, where 0+ means a valid value, negative invalid. Some models use counters, < 1 = empty / invalid, 1+ presence / valid.

The default could be further improved by e.g. adding a boolean `negate` plus some constants like `POSITIVE`, `NON_NEGATIVE`, `EMPTY` with the option to negate the meaning. That would cover more use-cases.
Do you think it is worth adding/changing ?